include/jemalloc/internal/arena_inlines_b.h

```diff
@@ -298,6 +298,47 @@ arena_dalloc_large(tsdn_t *tsdn, void *ptr, tcache_t *tcache, szind_t szind,
 	}
 }
 
+static inline bin_t *
+arena_get_bin(arena_t *arena, szind_t binind, unsigned binshard) {
+	bin_t *shard0 = (bin_t *)((uintptr_t)arena + arena_bin_offsets[binind]);
+	return shard0 + binshard;
+}
+
+JEMALLOC_ALWAYS_INLINE size_t
+arena_slab_regind_unlocked(div_info_t* div_info, szind_t binind,
```

Comment:
Let's refactor to avoid duplicate the logic for this function. You can add a impl function which is called by the original and this new one. The impl function will contain the shared logic and asserts.

Reply:
Since arena_slab_regind only contains one extra assert compared to arena_slab_regind_unlocked, I make arena_slab_regind directly call arena_slab_regind_unlocked to reduce the complexity. Does that work?

Reply:
Updated and changed the function name to arena_slab_regind_impl